src/Raven.Server/ServerWide/Commands/StartBucketMigrationCommand.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Linq;
+using Raven.Client.ServerWide;
+using Raven.Server.Utils;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.ServerWide.Commands
+{
+    public class StartBucketMigrationCommand : UpdateDatabaseCommand
+    {
+        public int SourceShard;
```

Comment:
readonly? private set?

Reply:
IIRC it can't be readonly due to the de-serializer